src/UI/Component/Dropzone/File/File.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace ILIAS\UI\Component\Dropzone\File;
+
+use ILIAS\UI\Component\Component;
+use ILIAS\UI\Component\Droppable;
+
+/**
+ * Interface File
+ *
+ * A dropzone where one can drop files on it to be uploaded on the server.
+ *
+ * @author  nmaerchy <nm@studer-raimann.ch>
+ *
+ * @package ILIAS\UI\Component\Dropzone\File
+ */
+interface File extends Component, Droppable {
+
+	/**
+	 * Get a dropzone like this where the files are uploaded to the given URL.
+	 *
+	 * @param string $url
+	 * @return $this
+	 */
+	public function withUploadUrl($url);
+
+	/**
+	 * Get the upload URL where the files are uploaded.
+	 *
+	 * @return string
+	 */
+	public function getUploadUrl();
+
+	/**
+	 * Get a dropzone like this only accepting the submitted file types for uploading, e.g.
+	 * ['jpg', 'png', 'gif'] to allow some image formats.
+	 *
+	 * @param array $types
+	 * @return $this
+	 */
+	public function withAllowedFileTypes(array $types);
+
+	/**
+	 * @return array
+	 */
+	public function getAllowedFileTypes();
+
+	/**
+	 * Get a dropzone like this, restricting the max number of files that can be uploaded.
+	 *
+	 * @param int $max
+	 * @return $this
+	 */
+	public function withMaxFiles($max);
+
+	/**
+	 * Get the max number of files that can be uploaded.
+	 *
+	 * @return int
+	 */
+	public function getMaxFiles();
+
+	/**
+	 * Get a dropzone like this, restricting the max file size of the files to the given limit (in bytes).
+	 *
+	 * @param int $limit Max size for any file uploaded in bytes
+	 * @return $this
+	 */
+	public function withFileSizeLimit($limit);
+
+	/**
+	 * Get the max file size limit in bytes.
+	 *
+	 * @return int
+	 */
+	public function getFileSizeLimit();
+
+	/**
+	 * Get a dropzone like this, allowing to set the filename for each file being uploaded.
+	 * The custom file name is sent as POST parameter along with the uploaded file.
+	 *
+	 * @param bool $state True to enable custom file names
+	 * @return $this
+	 */
+	public function withCustomFileNames($state);
```

Comment:
This is currently also possible in ILIAS: If you drop files over a course, they are listed in the right "panel" where you can enter a custom filename/description for the file object being created.